examples/hello-world/ml-to-fl/np/README.md

```diff
@@ -0,0 +1,111 @@
+
+```bash
+nvflare config -jt ../../../../job_templates/
+nvflare job list_templates
+nvflare job create -force -j ./jobs/np_param_full_transfer_full -w sag_np -sd ./code/ \
+-f config_fed_client.conf app_script=train_full.py params_transfer_type=FULL launch_once=false
+```
+
+Then we can run it using the NVFlare Simulator:
+
+```bash
+nvflare simulator -n 2 -t 2 ./jobs/np_param_full_transfer_full -w np_param_full_transfer_full_workspace
+```
+
+## Send model parameters differences back to the NVFlare server
+
+There are two ways to send model parameters differences back to the NVFlare server:
+
+1. Send the full parameters in training script, change params_transfer_type to "DIFF"
+2. Calculate the parameters differences in training script and send it back via "flare.send"
+
+For the first way, we can reuse the mock training script [./code/train_full.py](./code/train_full.py)
+
+But we need to pass different parameters when creating job:
+
+```bash
+nvflare job create -force -j ./jobs/np_param_full_transfer_diff -w sag_np -sd ./code/ \
+-f config_fed_client.conf app_script=train_full.py params_transfer_type=DIFF launch_once=false \
+-f config_fed_server.conf expected_data_kind=WEIGHT_DIFF
+```
+
+By setting "params_transfer_type=DIFF" we are using the NVFlare built-in parameter difference method to calculate differences.
+
+Then we can run it using the NVFlare Simulator:
+
+```bash
+nvflare simulator -n 2 -t 2 ./jobs/np_param_full_transfer_diff -w np_param_full_transfer_diff_workspace
+```
+
+For the second way, we write a new mock training script that calculate the model difference and send it back: [./code/train_diff.py](./code/train_diff.py)
+
+Note that we set the "params_type" to DIFF when creating flare.FLModel.
+
+Then we create the job using the following command:
+
+```bash
+nvflare job create -force -j ./jobs/np_param_diff_transfer_full -w sag_np -sd ./code/ \
+-f config_fed_client.conf app_script=train_diff.py launch_once=false \
+-f config_fed_server.conf expected_data_kind=WEIGHT_DIFF
+```
+
+The "params_transfer_type" is "FULL", means that we DO NOT calculate the difference again using the NVFlare built-in parameter difference method.
+
+Then we can run it using the NVFlare Simulator:
+
+```bash
+nvflare simulator -n 2 -t 2 ./jobs/np_param_diff_transfer_full -w np_param_diff_transfer_full_workspace
+```
+
+## Launch once for the whole job
+
+In some training scenarios, the data loading is taking a lot of time.
+And throughout the whole training job, we only want to load/set up the data once.
+
+In that case, we could use the "launch_once" option of "LauncherExecutor" and wraps our training script into a loop.
+
+We wrap the [./code/train_full.py](./code/train_full.py) into a loop: [./code/train_loop.py](./code/train_loop.py)
+
+Then we can create the job:
+
+```bash
+nvflare job create -force -j ./jobs/np_loop -w sag_np -sd ./code/ \
+-f config_fed_client.conf app_script=train_loop.py params_transfer_type=DIFF launch_once=true \
```

Comment:
`params_transfer_type` and `expected_data_kind` should be `FULL` and `WEIGHTS` for train_loop.py

Reply:
@SYangster good catch! fixed.